javaslang/src/main/java/javaslang/collection/Iterator.java

```diff
@@ -1392,24 +1392,37 @@ public T getNext() {
             final Iterator<T> that = this;
             return new AbstractIterator<T>() {
 
-                private T next = null;
+                private T next;
```

Comment:
> My plan is to resolve 68 issues (tendency increasing because of new tickets). 

I think our issues originate from more complicated code.

I have benchmarked the two alternatives like follows:
```java
public static class DropWhile extends Base {
    final Predicate<Integer> predicate = e -> e != ELEMENTS[CONTAINER_SIZE / 2];

    @Benchmark
    public int slang_persistent() {
        int length = Iterator.of(ELEMENTS).dropWhile(predicate).length();
        assert length == Array.of(ELEMENTS).dropWhile(predicate).length();
        return length;
    }
}
```
i.e. let's `drop` the iterator until the middle, and consume all resulting elements via `length`.

The results for `10`, `1_000` and `100_000` elements:
The explicit, long solution:
```java
Params          Score   Unit
    10  10,827,509.92  ops/s
  1000     144,264.98  ops/s
100000       1,333.39  ops/s
```

The reusing, short solution:
```java
Params          Score   Unit
    10   1,688,373.91  ops/s
  1000     191,893.42  ops/s
100000       2,065.07  ops/s
```

i.e. for very few elements, your solution is `10x` faster, after which the other one becomes faster.